constructr-machine/src/main/scala/de/heikoseeberger/constructr/machine/ConstructrMachine.scala

```diff
@@ -147,7 +147,11 @@ final class ConstructrMachine[A: Coordination.AddressSerialization, B <: Coordin
   }
 
   when(State.AddingSelf, coordinationTimeout) {
-    case Event(Coordination.SelfAdded(_), _) => goto(State.RefreshScheduled).using(stateData.copy(coordinationRetriesLeft = coordinationRetries))
+    case Event(Coordination.SelfAdded(context), _) =>
+      goto(State.RefreshScheduled).using(stateData.copy(
+        coordinationRetriesLeft = coordinationRetries,
+        context = context.asInstanceOf[B#Context]
```

Comment:
;-)
I think you should simply use a typed pattern in the pattern match three lines above: 

```
case Event(Coordination.SelfAdded(context: B#Context), _) => ...
```